components/button/button-split.js

```diff
@@ -0,0 +1,142 @@
+import './button.js';
+import '../colors/colors.js';
+import '../dropdown/dropdown.js';
+import '../dropdown/dropdown-menu.js';
+import '../icons/icon.js';
+import '../menu/menu.js';
+import { css, html, LitElement } from 'lit';
+import { FocusMixin } from '../../mixins/focus/focus-mixin.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+
+/**
+ * A split button that provides a main action button and slot for a menu.
+ */
+class ButtonSplit extends FocusMixin(LocalizeCoreElement(LitElement)) {
+
+	static get properties() {
+		return {
+			/**
+			 * ACCESSIBILITY: A description to be added to the main action button for accessibility when text does not provide enough context
+			 * @type {string}
+			 */
+			description: { type: String },
+			/**
+			 * Disables the main action and dropdown opener buttons
+			 * @type {boolean}
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * Tooltip text when disabled
+			 * @type {string}
+			 */
+			disabledTooltip: { type: String, attribute: 'disabled-tooltip' },
+			/**
+			 * Styles the buttons as a primary buttons
+			 * @type {boolean}
+			 */
+			primary: { type: Boolean, reflect: true },
+			/**
+			 * ACCESSIBILITY: REQUIRED: Accessible text for the main action button
+			 * @type {string}
+			 */
+			text: { type: String, reflect: true, required: true },
```

Comment:
Do you want to use `PropertyRequiredMixin` here so it'll throw if this isn't provided?

Reply:
Yep. With the current event API, we should probably make `key` required too eh?!?

Reply:
I'm not sure it's worth raising errors. They aren't going to get very far using the component without the `key`.